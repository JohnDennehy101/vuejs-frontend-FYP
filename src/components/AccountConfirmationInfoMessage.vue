<template>
  <div class="banner-message-container">
    <i data-testid="icon" class="fas fa-check-circle"></i>
    <p data-testid="infoMessage">
      {{ infoMessage }}
    </p>
  </div>
</template>

<script>
export default {
  props: {
    message: String,
  },
  data() {
    return {
      infoMessage: this.message,
    };
  },
};
</script>

<style scoped lang="scss">
.banner-message-container {
  display: flex;
  align-items: center;
  padding: 1rem;
}

p {
  margin: 0 0.5rem;
  color: #647196;
  font-size: 1rem;
}
svg {
  color: #647196;
  font-size: 1rem;
}
</style>
