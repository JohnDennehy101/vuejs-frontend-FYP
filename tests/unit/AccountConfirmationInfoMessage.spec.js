import { mount, shallowMount } from "@vue/test-utils";
import AccountConfirmationInfoMessage from "@/components/AccountConfirmationInfoMessage.vue";

describe("AccountConfirmationInfoMessage.vue", () => {
  afterEach(() => {
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const wrapper = shallowMount(AccountConfirmationInfoMessage, {
      props: {
        message:
          "Thanks for confirming your email. You will now be redirected to the home page",
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("checks that message is rendered correctly when passed as prop", async () => {
    const wrapper = mount(AccountConfirmationInfoMessage, {
      props: {
        message:
          "Thanks for confirming your email. You will now be redirected to the home page",
      },
    });

    const messageTextElement = wrapper.find('[data-testid="infoMessage"]');
    expect(messageTextElement.text()).toBe(
      "Thanks for confirming your email. You will now be redirected to the home page"
    );
  });

  it("checks that icon is rendered correctly", async () => {
    const wrapper = mount(AccountConfirmationInfoMessage, {
      props: {
        message:
          "Thanks for confirming your email. You will now be redirected to the home page",
      },
    });

    const iconElement = wrapper.find('[data-testid="icon"]');
    expect(iconElement.exists()).toBe(true);
  });
});
