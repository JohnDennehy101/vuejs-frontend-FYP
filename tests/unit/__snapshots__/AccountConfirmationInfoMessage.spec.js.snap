// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountConfirmationInfoMessage.vue should render correctly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="banner-message-container"
      >
        <i
          class="fas fa-check-circle"
          data-testid="icon"
        />
        <p
          data-testid="infoMessage"
        >
          Thanks for confirming your email. You will now be redirected to the home page
        </p>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "VTU_COMPONENT": Object {
          "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
          "message": "Thanks for confirming your email. You will now be redirected to the home page",
        },
      },
      "da": null,
      "data": Object {},
      "devtoolsRawSetupState": Object {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": Object {
            "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
            "message": "Thanks for confirming your email. You will now be redirected to the home page",
          },
          "_value": Object {
            "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
            "message": "Thanks for confirming your email. You will now be redirected to the home page",
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "VTU_COMPONENT": Object {
          "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
          "message": "Thanks for confirming your email. You will now be redirected to the home page",
        },
      },
      "refs": Object {
        "VTU_COMPONENT": Object {
          "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
          "message": "Thanks for confirming your email. You will now be redirected to the home page",
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": Array [],
        "detached": true,
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": Object {
        "VTU_COMPONENT": Object {
          "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
          "message": "Thanks for confirming your email. You will now be redirected to the home page",
        },
      },
      "slots": Object {},
      "slotsProxy": null,
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "__v_isReadonly": 0,
            "__v_isRef": 0,
            "__v_isShallow": 0,
            "__v_raw": 0,
            "__v_skip": 4,
            "_isMockFunction": 0,
            "constructor": 0,
            "infoMessage": 2,
            "isVueInstance": 0,
            "message": 3,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
            "message": "Thanks for confirming your email. You will now be redirected to the home page",
          },
          "da": null,
          "data": Object {
            "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "message": "Thanks for confirming your email. You will now be redirected to the home page",
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "message": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
            "message": "Thanks for confirming your email. You will now be redirected to the home page",
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": Array [],
            "detached": true,
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "slotsProxy": null,
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "ce": undefined,
                "children": null,
                "component": null,
                "ctx": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <i
                  class="fas fa-check-circle"
                  data-testid="icon"
                />,
                "key": null,
                "patchFlag": -1,
                "props": Object {
                  "class": "fas fa-check-circle",
                  "data-testid": "icon",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "i",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Thanks for confirming your email. You will now be redirected to the home page",
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p
                  data-testid="infoMessage"
                >
                  Thanks for confirming your email. You will now be redirected to the home page
                </p>,
                "key": null,
                "patchFlag": 1,
                "props": Object {
                  "data-testid": "infoMessage",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Thanks for confirming your email. You will now be redirected to the home page",
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p
                  data-testid="infoMessage"
                >
                  Thanks for confirming your email. You will now be redirected to the home page
                </p>,
                "key": null,
                "patchFlag": 1,
                "props": Object {
                  "data-testid": "infoMessage",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="banner-message-container"
            >
              <i
                class="fas fa-check-circle"
                data-testid="icon"
              />
              <p
                data-testid="infoMessage"
              >
                Thanks for confirming your email. You will now be redirected to the home page
              </p>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "class": "banner-message-container",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "data": [Function],
            "props": Object {
              "message": [Function],
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="banner-message-container"
        >
          <i
            class="fas fa-check-circle"
            data-testid="icon"
          />
          <p
            data-testid="infoMessage"
          >
            Thanks for confirming your email. You will now be redirected to the home page
          </p>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "message": "Thanks for confirming your email. You will now be redirected to the home page",
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "data": [Function],
          "props": Object {
            "message": [Function],
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="banner-message-container"
        >
          <i
            class="fas fa-check-circle"
            data-testid="icon"
          />
          <p
            data-testid="infoMessage"
          >
            Thanks for confirming your email. You will now be redirected to the home page
          </p>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.4",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": Object {
    "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
    "message": "Thanks for confirming your email. You will now be redirected to the home page",
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {
      "infoMessage": "Thanks for confirming your email. You will now be redirected to the home page",
      "message": "Thanks for confirming your email. You will now be redirected to the home page",
    },
  },
  "wrapperElement": <div
    class="banner-message-container"
  >
    <i
      class="fas fa-check-circle"
      data-testid="icon"
    />
    <p
      data-testid="infoMessage"
    >
      Thanks for confirming your email. You will now be redirected to the home page
    </p>
  </div>,
}
`;
